.circleci/config.yml

```diff
@@ -1,141 +1,66 @@
-# Java Maven CircleCI 2.0 configuration file
-#
-# Check https://circleci.com/docs/2.0/language-java/ for more details
-#
-version: 2.0
-
-#########################
-# CACHES
-#########################
-
-restore_cache: &restore_m2_cache
-  keys:
-    - m2-{{ checksum "pom.xml" }}
-    - m2- # used if checksum fails
-
-#########################
-# COMMANDS
-#########################
-
-commands:
-
-  restore_m2_cache:
-    description: "Restore m2 cache for specified maven task"
-    parameters:
-      cache_name:
-        default: ""
-        type: enum
-        enum: ["", "security-proxy-request-mapper"]
-    steps:
-      - restore_cache:
-          keys:
-            - m2-<<parameters.cache_name>>-{{ checksum "pom.xml" }}
-            - m2-<<parameters.cache_name>>- # used if checksum fails
-            - m2-
-
-  save_m2_cache:
-    description: "Save m2 cache for specified maven task"
-    parameters:
-      cache_name:
-        default: ""
-        type: enum
-        enum: ["", "security-proxy-request-mapper"]
-    steps:
-      - save_cache:
-          key: m2-<<parameters.cache_name>>-{{ checksum "pom.xml" }}
-          paths:
-            - ~/.m2
+version: 2
 
 jobs:
-
   #########################
   # BUILD
   #########################
   build:
+    working_directory: ~/repo
     docker:
       - image: circleci/openjdk:8-jdk
-    working_directory: ~/repo
+        environment:
```

Comment:
You have now specified the environment variables in the container. I don't know what the difference in this case is but I wonder whether we should set them in the container vs in the job.

Reply:
It is done that way because defining of env variables on container level is the easiest way to deal with env variables in circle ci (for sure easiest than that https://circleci.com/docs/2.0/env-vars/ and more readable)

Reply:
👍cool. I didn't know that.